letsencrypt/client/account.py

```diff
@@ -0,0 +1,205 @@
+"""Creates ACME accounts for server."""
+import logging
+import os
+import re
+
+import configobj
+import zope.component
+
+from letsencrypt.acme import messages2
+
+from letsencrypt.client import crypto_util
+from letsencrypt.client import errors
+from letsencrypt.client import interfaces
+from letsencrypt.client import le_util
+
+from letsencrypt.client.display import util as display_util
+
+
+class Account(object):
+    """ACME protocol registration.
+
+    :ivar config: Client configuration object
+    :type config: :class:`~letsencrypt.client.interfaces.IConfig`
+    :ivar key: Account/Authorized Key
+    :type key: :class:`~letsencrypt.client.le_util.Key`
+
+    :ivar str email: Client's email address
+    :ivar str phone: Client's phone number
+
+    :ivar regr: Registration Resource
+    :type regr: :class:`~letsencrypt.acme.messages2.RegistrationResource`
+
+    """
+
+    # Just make sure we don't get pwned
+    # Make sure that it also doesn't start with a period or have two consecutive
+    # periods <- this needs to be done in addition to the regex
+    EMAIL_REGEX = "[a-zA-Z0-9._%+-]+@[a-zA-Z0-9.-]+$"
+
+    def __init__(self, config, key, email=None, phone=None, regr=None):
+        le_util.make_or_verify_dir(
+            config.accounts_dir, 0o700, os.geteuid())
+        self.key = key
+        self.config = config
+        if email is not None and self.safe_email(email):
+            self.email = email
+        else:
+            self.email = None
+        self.phone = phone
+
+        self.regr = regr
+
+    @property
+    def uri(self):
+        """URI link for new registrations."""
+        if self.regr is not None:
+            return self.regr.uri
+
+    @property
+    def new_authzr_uri(self):  # pylint: disable=missing-docstring
+        if self.regr is not None:
+            if self.regr.new_authzr_uri:
+                return self.regr.new_authzr_uri
+            else:
+                # Default: spec says they "may" provide the header
+                # ugh.. acme-spec #93
+                return "https://%s/acme/new-authz" % self.config.server
```

Comment:
Sorry.. missed the second comment... I did realize that the code was unreachable and meant to fix the networking code.  The spec states that it is optional... which breaks technically valid servers...

I think this would be a dumb thing to argue though... really I think the spec should just be fixed.  will remove.